src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2691,6 +2697,12 @@ SqlTaskList(Job *job)
 	List *fragmentCombinationList = FragmentCombinationList(rangeTableFragmentsList,
 															jobQuery, dependentJobList);
 
+	/* Adjust RelabelType and CoerceViaIO nodes that are improper for deparsing */
+	if (QueryTreeHasRelabelTypeOrCoerceViaIONodes((Node *) jobQuery))
+	{
```

Comment:
would be nice to have a comment on why we first check the query tree and then only do AdjustImproperForDeparseNodes later. It might not be obvious for the readers.